### PR TITLE
feat: add hanson metric

### PR DESCRIPTION
**Pull request by alex-pirogov**

### CODE REVIEW

File: src/metrics/HansonMetric.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import McCabeCC from './McCabeCC';
+import NumberOfHansonOperators from './NumberOfHansonOperators';
+// @ts-ignore
+import * as Styx from 'styx';
+
+export default class HansonMetric implements IMetric {
+  private _name = 'Hanson metric';
+  private _info =
+    'Complexity metric defined by Hanson https://dl.acm.org/doi/pdf/10.1145/954373.954375';
+  private _scope = 'any';
```
Comment (IntegralTeam) [inline src/metrics/HansonMetric.ts:12]: Are you sure that the scope is any?? How you gonna calculate cyclomatic complexity for the class or group of classes? Will it work?

Comment (andrey-kulevich) [inline src/metrics/HansonMetric.ts]: Are you sure that the scope is 'any'? How you gonna calculate cyclomatic complexity for the entire class or the group of classes?

File: src/metrics/HansonMetric.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  public get scope() {
+    return this._scope as any;
+  }
+
+  public run(program: esprima.Program, programFlow: typeof Styx.parse) {
+    return {
+      value: {
+        A: new McCabeCC().run(programFlow).value,
+        B: new NumberOfHansonOperators().run(program),
```
Comment (andrey-kulevich) [inline src/metrics/HansonMetric.ts:30]: I have this metric already, TotalNumberOfOperators.ts, use it

File: src/metrics/McCabeCC.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return this._scope as any;
+  }
+
+  public run(programFlow: typeof Styx.parse) {
+    const m = programFlow.flowGraph.edges.length,
+      n = programFlow.flowGraph.nodes.length;
+
+    return {
+      m,
+      n,
```
Comment (andrey-kulevich) [inline src/metrics/McCabeCC.ts:29]: Why do you return m, n?

Comment (alex-pirogov) [reply src/metrics/McCabeCC.ts:29]: i think this values might be useful for other metrics, they are essential for mc cabe cc

**Pull request merged**